### ISSUE TITLE
vcsim does not implement support for VSLM ExtendDisk

### ISSUE THREAD

**Issue by protochron** [vmware/govmomi#2289]
It seems like govmomi supports most of the VSLM APIs, but it looks like vcsim is missing `ExtendDisk`. Calling `ExtendDisk` causes vcsim to emit a message: 
```
VcenterVStorageObjectManager:VStorageObjectManager does not implement: HostExtendDisk_Task
```
The code I'm using looks something like: 
```go
manager := vslm.NewObjectManager(vsClient)
task, err := manager.ExtendDisk(ctx, ds.Reference(), uuid, size*1024)
if err != nil {
	return err
}

_, err = task.WaitForResult(ctx, nil)
if err != nil {
	return err
}
```

It seems like something vcsim should support unless there's a deeper reason why it's not implemented in any of the simulator code.

**Comment by dougm** [comment]
We can support it.. haven't tested yet, but something like this should work:
```diff
modified   simulator/vstorage_object_manager.go
@@ -279,6 +279,23 @@ func (m *VcenterVStorageObjectManager) CreateDiskTask(req *types.CreateDisk_Task
 	}
 }
 
+func (m *VcenterVStorageObjectManager) ExtendDiskTask(req *types.ExtendDisk_Task) soap.HasFault {
+	task := CreateTask(m, "extendDisk", func(*Task) (types.AnyType, types.BaseMethodFault) {
+		obj := m.object(req.Datastore, req.Id)
+		if obj == nil {
+			return nil, &types.InvalidArgument{}
+		}
+
+		return nil, nil
+	})
+
+	return &methods.ExtendDisk_TaskBody{
+		Res: &types.ExtendDisk_TaskResponse{
+			Returnval: task.Run(),
+		},
+	}
+}
+
 func (m *VcenterVStorageObjectManager) DeleteVStorageObjectTask(req *types.DeleteVStorageObject_Task) soap.HasFault {
 	task := CreateTask(m, "deleteDisk", func(*Task) (types.AnyType, types.BaseMethodFault) {
 		obj := m.object(req.Datastore, req.Id)
```

**Comment by dougm** [comment]
Fixed by #2481
